tests/test_template_types.py

```diff
@@ -819,6 +819,35 @@ def test_subject_line_gets_replaced():
     assert template.subject == 'Jo'
 
 
+@pytest.mark.parametrize("content, values, expected_count", [
+    ("Content with ((placeholder))", {"placeholder": "something extra"}, 28),
+    ("Content with ((placeholder))", {"placeholder": ""}, 12),
+    ("((placeholder))  ", {"placeholder": "  "}, 0),
+    ("  ", {}, 0),
+])
+def test_WithSubjectTemplate_character_count(content, values, expected_count):
+    template = WithSubjectTemplate({"content": content, 'subject': 'Hi'})
+    template.values = values
+    assert template.content_count == expected_count
+
+
+@pytest.mark.parametrize("content, values, prefix, expected_count", [
+    ("Content with ((placeholder))", {"placeholder": "something extra"}, None, 28),
+    ("Content with ((placeholder))", {"placeholder": ""}, None, 12),
+    ("Just content", {}, None, 12),
+    ("((placeholder))  ", {"placeholder": "  "}, None, 0),
+    ("  ", {}, None, 0),
+    ("Content with ((placeholder))", {"placeholder": "something extra"}, "GDS", 33),
+    ("Just content", {}, "GDS", 12),
+    ("((placeholder))  ", {"placeholder": "  "}, "GDS", 5),
+    ("  ", {}, "GDS", 5),
+])
+def test_SMSMessageTemplate_character_count(content, values, prefix, expected_count):
```

Comment:
Turns out we already have a test for most of this (although I haven't compared the test cases themselves so there could be some that are missing).

Take a look at `test_get_character_count_of_content` in `test_base_template.py`

Reply:
ahh, thanks, I will check 👍